include/envoy/http/header_map.h

```diff
@@ -310,6 +310,7 @@ class HeaderEntry {
   HEADER_FUNC(OtSpanContext)                                                                       \
   HEADER_FUNC(Origin)                                                                              \
   HEADER_FUNC(Path)                                                                                \
+  HEADER_FUNC(OriginalPathForUpstream)                                                             \
```

Comment:
Do we need a new header? If the path is change can we still just use EnvoyOriginalPath above?

Reply:
Ah, the EnvoyOriginalPath, I was curious about using that.  The issue as I understand it, is that ideally keeping everything "backward compatible" was a high priority.  So leaving normalize_path in situ as is, with minimal processing changes presented a lower risk option.  I'd prefer to re-use what code variables already exist rather than adding a new header.  If we can re-use the EnvoyOriginalPath for this then all the better.  I will dig into it's usage.  If I understand the requirement correctly, it would seem that once filtering and routing phase is complete then the could be set to the upstream normalized/non-normalized value depending on settings.  The caveat here is extra checking of the OriginalEnvoyPath might be needed for any code that relies on the original usage of the Path variable (will verify usage).

Reply:
One thing I worry about is potential misuse and inconsistent use by filters. Ideally we only make use of the original path when sending to upstream..

Reply:
Yeah I think we should use a single header here. Re: back compat, as this is a new feature, and controlled by an enum, IMO we can make it clear what that header holds in each enum case.